### PR TITLE
Parse cluster proxy & trust variables

### LINKED ISSUE #1772: Cluster variables for proxy and custom certs
We have consensus on cluster vars for proxy and custom certs. These variables should replace whatever is being currently to set proxy and certificates in KappControllerConfig, VsphereCPIConfig and VsphereCSIConfig

```
    - name: proxy
      required: false
      schema:
        openAPIV3Schema:
          type: object
          properties:
            httpProxy:
              type: string
            httpsProxy:
              type: string
            noProxy:
              type: string
    - name: trust
      required: false
      schema:
        openAPIV3Schema:
          type: object
          properties:
            additionalTrustedCAs:
              type: array
              items:
                type: object
                properties:
                  name:
                    type: string
                  data:
                    type: string
```

### PR DESCRIPTION
**Pull request by ggpaue**
### What this PR does / why we need it
parse cluster proxy & trust variables
### Which issue(s) this PR fixes

Fixes #1772 

### Describe testing done for PR

1. modified testdata for bootstrapcontroller and kapp controller
2. tested with Ginkgo local testing and verified variables can be parsed

### Release note

```release-note

```

### PR Checklist

- [ ] Squash the commits into one or a small number of logical commits
      
- [ ] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [ ] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: addons/pkg/util/cluster_variable_util.go [hunk 1]
```diff
@@ -29,6 +30,41 @@ func ParseClusterVariableString(cluster *clusterapiv1beta1.Cluster, variableName
...
+	var result interface{}
+
+	result, err := parseClusterVariable(cluster, variableName)
+	if err != nil || result == nil {
+		return "", err
+	}
+	interfaceVars := result.(map[string]interface{})
+
+	if valueName, ok := interfaceVars[keyName]; ok {
+		return valueName.(string), err
```
Comment (blc1996) [inline addons/pkg/util/cluster_variable_util.go:43]: Can we add a type check before return `valueName.(string)`? Possibly valueName might not be a string?

File: addons/pkg/util/cluster_variable_util.go [hunk 2]
```diff
@@ -29,6 +30,41 @@ func ParseClusterVariableString(cluster *clusterapiv1beta1.Cluster, variableName
...
+    }
+	return "", err
+}
+
+func ParseClusterVariableList(cluster *clusterapiv1beta1.Cluster, variableName, keyName, data string) (string, error) {
+	var result interface{}
+    var sb strings.Builder
```
Comment (blc1996) [inline addons/pkg/util/cluster_variable_util.go:50]: indentation seems wrong.

File: addons/pkg/util/cluster_variable_util.go [hunk 3]
```diff
@@ -29,6 +30,41 @@ func ParseClusterVariableString(cluster *clusterapiv1beta1.Cluster, variableName
...
+
+	result, err := parseClusterVariable(cluster, variableName)
+	if err != nil || result == nil {
+		return "", err
+	}
+	interfaceVars := result.(map[string]interface{})
+	if valueList, ok := interfaceVars[keyName]; ok {
+		rec := valueList.([]interface{})
```
Comment (blc1996) [inline addons/pkg/util/cluster_variable_util.go:58]: Same here, we need to check if `valueList.([]interface{})` conversion is valid.

**Review by blc1996 — COMMENTED** [review]
Can we also take care of the changes in kapp-controller config controller for skipTLSVerify?
https://github.com/vmware-tanzu/tanzu-framework/blob/main/addons/controllers/kapp-controller/kappcontrollerconfig_types.go#L115-L116

File: addons/pkg/util/cluster_variable_util.go (hunk 1 above)
Comment (ggpaue) [reply addons/pkg/util/cluster_variable_util.go]: added type check before return

File: addons/pkg/util/cluster_variable_util.go (hunk 3 above)
Comment (ggpaue) [reply addons/pkg/util/cluster_variable_util.go]: added type check before return

**Comment by ggpaue** [comment]
> Can we also take care of the changes in kapp-controller config controller for skipTLSVerify? https://github.com/vmware-tanzu/tanzu-framework/blob/main/addons/controllers/kapp-controller/kappcontrollerconfig_types.go#L115-L116

Ah I added it in clusterbootstrap controller, let me move it to kapp-controller

**Review by blc1996 — APPROVED** [review]
LGTM

**Review by shivaani0505 — APPROVED** [review]
LGTM

**Pull request merged**